### PR TITLE
added soft-char limit widget to product category name

### LINKED ISSUE #7789: Increase character limit for PNI category names
sync_git is failing with French translations of some PNI product category names. I’d suggest we set that limit to 100 chars to keep a margin.

```
INFO - Pull: Importing changes in file 'locales/fr/snippets/wagtailpages.BuyersGuideProductCategory/117-wearables-smart-glasses-ar-headsets-sort-order-1.po'
Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(50)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "network-api/manage.py", line 34, in <module>
    execute_from_command_line(sys.argv)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/app/.heroku/python/lib/python3.7/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/core/management/base.py", line 371, in execute
    output = self.handle(*args, **options)
  File "/app/.heroku/python/lib/python3.7/site-packages/wagtail_localize_git/management/commands/sync_git.py", line 19, in handle
    SyncManager(logger=logger).sync()
  File "/app/.heroku/python/lib/python3.7/site-packages/wagtail_localize_git/sync.py", line 165, in sync
    _pull(repo, self.logger)
  File "/app/.heroku/python/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/app/.heroku/python/lib/python3.7/site-packages/wagtail_localize_git/sync.py", line 39, in _pull
    importer.import_resource(translation, po)
  File "/app/.heroku/python/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/app/.heroku/python/lib/python3.7/site-packages/wagtail_localize_git/importer.py", line 29, in import_resource
    translation.save_target()
  File "/app/.heroku/python/lib/python3.7/site-packages/wagtail_localize/models.py", line 1241, in save_target
    copy_parent_pages=True,
  File "/app/.heroku/python/lib/python3.7/site-packages/wagtail_localize/models.py", line 782, in create_or_update_translation
    translation.save()
  File "/app/network-api/networkapi/wagtailpages/pagemodels/products.py", line 204, in save
    super().save(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/models/base.py", line 754, in save
    force_update=force_update, update_fields=update_fields)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/models/base.py", line 792, in save_base
    force_update, using, update_fields,
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/models/base.py", line 873, in _save_table
    forced_update)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/models/base.py", line 926, in _do_update
    return filtered._update(values) > 0
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/models/query.py", line 803, in _update
    return query.get_compiler(self.db).execute_sql(CURSOR)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 1522, in execute_sql
    cursor = super().execute_sql(result_type)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 1156, in execute_sql
    cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.7/site-packages/sentry_sdk/integrations/django/__init__.py", line 500, in execute
    return real_execute(self, sql, params)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/app/.heroku/python/lib/python3.7/site-packages/scout_apm/django/instruments/sql.py", line 59, in db_execute_hook
    return execute(sql, params, many, context)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/app/.heroku/python/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.DataError: value too long for type character varying(50)

```

### PR DESCRIPTION
**Pull request by danielfmiranda**
Closes #7789

**Link to sample test page**: https://foundation-s-7789-exten-vidgun.herokuapp.com/en/cms/snippets/wagtailpages/buyersguideproductcategory/edit/3/

**Steps to test**:

1. Visit the edit page linked above ([review app details](https://mozilla.slack.com/archives/CBKP2SG94/p1639007605007600)):
2. Note the addition of the character limit widget on the bottom right of the "name" field. It should work as expected however since this is a soft limit, it will still let you save if you go over the limit.
3. If everything is working as expected, testing is complete!

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 1]
```diff
@@ -170,7 +171,10 @@ class BuyersGuideProductCategory(index.Indexed, TranslatableMixin, LocalizedSnip
     )
 
     panels = [
-        FieldPanel('name'),
+        FieldPanel(
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:174]: Wanted to note that the limit for this field was already set to 100, so I just had to add the widget

**Review by richbrennan — APPROVED** [review]
Looks good, thanks Daniel!

If you did want a hard limit on the field you could probably override the [Wagtail](https://docs.wagtail.io/en/stable/advanced_topics/customisation/page_editing_interface.html#wagtail.admin.forms.WagtailAdminModelForm) `WagtailAdminModelForm` for the Snippet.
Shout if you'd like more info.

**Review by TheoChevalier — CHANGES_REQUESTED** [review]
Hm, looks like the limit needs to be updated somewhere else, because editing a translation with 50+ characters (e.g. adding a couple chars to https://foundation.mozilla.org/cms/snippets/wagtailpages/buyersguideproductcategory/edit/168/) fails with the following:


`2021-12-09T17:38:39.565531+00:00 app[web.6]: django.db.utils.DataError: value too long for type character varying(50)`

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 2]
```diff
@@ -152,7 +153,8 @@ class BuyersGuideProductCategory(index.Indexed, TranslatableMixin, LocalizedSnip
 
     slug = models.SlugField(
         blank=True,
-        help_text='A URL-friendly version of the category name. This is an auto-generated field.'
+        help_text='A URL-friendly version of the category name. This is an auto-generated field.',
+        max_length=100
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:157]: Hi @TheoChevalier! It turns out the error was being caused by the slug field, according to the django docs, the default max length for a slug field is 50! Updating this and migrating on my local machine solved the issue

Comment (TheoChevalier) [reply network-api/networkapi/wagtailpages/pagemodels/products.py:157]: Thanks Daniel, nice find :)

**Review by TheoChevalier — APPROVED** [review]

**Pull request merged**